modeling/data.go

```diff
@@ -51,42 +42,96 @@ type EndpointSliceGroup struct {
 	Label string
 	// Composition stores contribution of endpoints in this group from different
 	// zones
-	Composition map[string]int
-	// Weights of endpoints in this group for different zones while routing
-	Weights map[string]float64
+	Composition map[string]weightedEndpoints
+	// ZoneTrafficWeights this sliceGroup has for requests from different zones
+	ZoneTrafficWeights map[string]float64
 }
 
-// Traffic represents the detailed traffic infomation of a zone
-type Traffic struct {
+// SimulationResult is to collect metrics of a simulation result
+type SimulationResult struct {
+	// InZoneTraffic is the total ratio of traffic that stays in the same zone
+	InZoneTraffic float64
+	// TrafficDistribution groups Traffic by zone name
+	TrafficDistribution map[string]zoneTraffic
+	// MaxDeviation of traffic load of all endpoints
+	MaxDeviation float64
+	// MeanDeviation of traffic load of all endpoints
+	MeanDeviation float64
+	// DeviationSD abbr of standard deviation of the daviation of traffic load
+	// of all endpoints
+	DeviationSD float64
+}
+
+type regionInfo struct {
+	// Total number of nodes of all zones
+	totalNodes int
+	// Total number of endpoints of all zones
+	totalEndpoints int
+	// Detailed information of each zone, zone name - exact zone
+	zoneDetails map[string]Zone
+}
+
+// endpoints with weights are used to do routing inside an EndpointSliceGroup
+type weightedEndpoints struct {
+	// Number of endpoints
+	Number int
```

Comment:
Does this field need to be exported? Seems unlikely given that the enclosing type is not exported.

Reply:
Related: The `modeling` package is getting quite large. It may make sense to split out some parts like simulation into different packages. That doesn't necessarily have to happen yet, but if you were already about to, you could export any types that might be needed by other packages.

Reply:
fixed. That makes sense, could do this after this PR, it should involve some work to split the package reasonably